libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -198,21 +198,11 @@ private fun DirectionsResponse.Builder.updateWaypoints(
     oldWaypoints: List<DirectionsWaypoint>?,
     updatedWaypoints: List<DirectionsWaypoint?>?,
 ): DirectionsResponse.Builder {
-    if (oldWaypoints == null) {
+    if (oldWaypoints == null || updatedWaypoints == null) {
```

Comment:
Should we check for emptiness as well? I mean what if the request adds an empty array instead of not adding it at all? We'll forget about this place. I think it would make sense that empty waypoints should not be refreshed.

Reply:
The way we process the information handles this case (it iterates over `oldWaypoints`), we also have unit tests that cover this case.